media/server/gstplayer/source/tasks/generic/SetVolume.cpp

```diff
@@ -35,15 +43,75 @@ SetVolume::~SetVolume()
     RIALTO_SERVER_LOG_DEBUG("SetVolume finished");
 }
 
+firebolt::rialto::wrappers::rgu_Ease convertEaseTypeToRguEase(EaseType easeType)
+{
+    switch (easeType)
+    {
+    case EaseType::EASE_LINEAR:
+        return firebolt::rialto::wrappers::rgu_Ease::EaseLinear;
+    case EaseType::EASE_IN_CUBIC:
+        return firebolt::rialto::wrappers::rgu_Ease::EaseInCubic;
+    case EaseType::EASE_OUT_CUBIC:
+        return firebolt::rialto::wrappers::rgu_Ease::EaseOutCubic;
+    default:
+        RIALTO_SERVER_LOG_ERROR("Unknown EaseType, defaulting to EaseLinear");
+        return firebolt::rialto::wrappers::rgu_Ease::EaseLinear;
+    }
+}
+
 void SetVolume::execute() const
 {
     RIALTO_SERVER_LOG_DEBUG("Executing SetVolume");
+
     if (!m_context.pipeline)
     {
         RIALTO_SERVER_LOG_ERROR("Setting volume failed. Pipeline is NULL");
         return;
     }
-    m_gstWrapper->gstStreamVolumeSetVolume(GST_STREAM_VOLUME(m_context.pipeline), GST_STREAM_VOLUME_FORMAT_LINEAR,
-                                           m_volume);
+    bool isNormalVolumeChange = (m_volumeDuration == 0);
+    GstElement *audioSink = m_player.getSink(firebolt::rialto::MediaSourceType::AUDIO);
+
+    if (!isNormalVolumeChange && audioSink &&
```

Comment:
I would change the variable name to isImmediateVolumeChange
I would process the immediate volume change as first one. In my opinion it will be the one, which will be the most frequently used and the code will be more readable.

if (isImmediateVolumeChange)
...
else if (audioSink &&  m_glibWrapper->gObjectClassFindProperty(G_OBJECT_GET_CLASS(audioSink), "audio-fade"))
...
else if (m_rdkGstreamerUtilsWrapper->isSocAudioFadeSupported())
..
else
WRN log